src/Microsoft.OData.Core/ODataWriter.cs

```diff
@@ -324,7 +324,7 @@ public ODataWriter Write(ODataProperty primitiveProperty, Action nestedAction)
         /// <summary> Asynchronously write a primitive property within a resource. </summary>
         /// <returns>A task instance that represents the asynchronous write operation.</returns>
         /// <param name="primitiveProperty">The primitive property to write.</param>
-        public virtual Task WriteStartAsync(ODataProperty primitiveProperty)
+        public virtual Task WriteStartAsync(ODataPropertyInfo primitiveProperty)

```

Comment:
>ODataPropertyInfo [](start = 44, length = 17)

ODataPropertyInfo is the base class of ODataProperty, So it's not a breaking change, right?

How about other Write(...) method?  For example: Line 315?

Reply:
Right.

---
In reply to: [427636669](https://github.com/OData/odata.net/pull/1770#discussion_r427636669) [](ancestors = 427636669)

Reply:
In 315, we are writing the entire property (name and value) so it needs the property, not just a property info.

---
In reply to: [428232034](https://github.com/OData/odata.net/pull/1770#discussion_r428232034) [](ancestors = 428232034,427636669)